### PR TITLE
C# 9 reference update: a with expression

### PR DESCRIPTION
**Pull request by pkulikov**
Contributes to #20585

### CODE REVIEW

File: docs/csharp/language-reference/operators/index.md [hunk 1]
```diff
@@ -75,6 +75,7 @@ The following table lists the C# operators starting with the highest precedence
 | [+x](arithmetic-operators.md#unary-plus-and-minus-operators), [-x](arithmetic-operators.md#unary-plus-and-minus-operators), [\!x](boolean-logical-operators.md#logical-negation-operator-), [~x](bitwise-and-shift-operators.md#bitwise-complement-operator-), [++x](arithmetic-operators.md#increment-operator-), [--x](arithmetic-operators.md#decrement-operator---), [^x](member-access-operators.md#index-from-end-operator-), [(T)x](type-testing-and-cast.md#cast-expression), [await](await.md), [&x](pointer-related-operators.md#address-of-operator-), [*x](pointer-related-operators.md#pointer-indirection-operator-), [true and false](true-false-operators.md) | Unary |
 | [x..y](member-access-operators.md#range-operator-) | Range |
 | [switch](switch-expression.md) | `switch` expression |
+| [with](with-expression.md) | `with` expression |
```
Comment (pkulikov) [inline docs/csharp/language-reference/operators/index.md:78]: This precedence table update is according to the spec proposal: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/records#with-expression
Is that correct understanding?

File: docs/csharp/language-reference/operators/operator-overloading.md [hunk 2]
```diff
@@ -38,7 +38,7 @@ The following table provides information about overloadability of C# operators:
...
+|[^x](member-access-operators.md#index-from-end-operator-), [x = y](assignment-operator.md), [x.y](member-access-operators.md#member-access-expression-), [`x?.y`](member-access-operators.md#null-conditional-operators--and-), [c ? t : f](conditional-operator.md), [x ?? y](null-coalescing-operator.md), [x ??= y](null-coalescing-operator.md), [x..y](member-access-operators.md#range-operator-), [x->y](pointer-related-operators.md#pointer-member-access-operator--), [=>](lambda-operator.md), [f(x)](member-access-operators.md#invocation-expression-), [as](type-testing-and-cast.md#as-operator), [await](await.md), [checked](../keywords/checked.md), [unchecked](../keywords/unchecked.md), [default](default.md), [delegate](delegate-operator.md), [is](type-testing-and-cast.md#is-operator), [nameof](nameof.md), [new](new-operator.md), [sizeof](sizeof.md), [stackalloc](stackalloc.md), [switch](switch-expression.md), [typeof](type-testing-and-cast.md#typeof-operator), [with](with-expression.md)|These operators cannot be overloaded.|
```
Comment (pkulikov) [inline docs/csharp/language-reference/operators/operator-overloading.md:41]: `with` is not overloadable

File: docs/csharp/language-reference/operators/with-expression.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+---
+title: "with expression - C# reference"
+description: "Learn about a with expression that performs non-destructive mutation of C# records"
+ms.date: 11/10/2020
+f1_keywords:
+  - "with_CSharpKeyword"
```
Comment (pkulikov) [inline docs/csharp/language-reference/operators/with-expression.md:6]: @Youssef1313 I've noticed you've worked recently in this area. Is that F1 keyword supported or going to be supported (so, I can leave it here)?

Comment (Youssef1313) [reply docs/csharp/language-reference/operators/with-expression.md:6]: I think it should work. But added a test for it in #49236 to confirm.

Comment (pkulikov) [reply docs/csharp/language-reference/operators/with-expression.md:6]: @Youssef1313 thank you
dotnet/roslyn#49236

File: docs/csharp/whats-new/csharp-9.md [hunk 4]
```diff
@@ -92,11 +92,13 @@ The compiler produces a `Deconstruct` method for positional records. The `Decons
...
 
-Finally, records support ***with-expressions***. A ***with-expression*** instructs the compiler to create a copy of a record, but *with* specified properties modified:
+Finally, records support [`with` expressions](../language-reference/operators/with-expression.md). A ***`with` expression*** instructs the compiler to create a copy of a record, but *with* specified properties modified:
 
 :::code language="csharp" source="snippets/whats-new-csharp9/PositionalRecords.cs" ID="Wither":::
 
-The above line creates a new `Person` record where the `LastName` property is a copy of `person`, and the `FirstName` is "Paul". You can set any number of properties in a with-expression.  Any of the synthesized members except the "clone" method may be written by you. If a record type has a method that matches the signature of any synthesized method, the compiler doesn't synthesize that method. The earlier `Dog` record example contains a hand coded <xref:System.String.ToString> method as an example.
+The above line creates a new `Person` record where the `LastName` property is a copy of `person`, and the `FirstName` is "Paul". You can set any number of properties in a `with` expression.
```
Comment (Youssef1313) [inline docs/csharp/whats-new/csharp-9.md:99]: "Paul" might get localized unintentionally.

Comment (pkulikov) [reply docs/csharp/whats-new/csharp-9.md]: Added the backticks then.

File: docs/csharp/language-reference/operators/index.md (hunk 1 above)
Comment (BillWagner) [reply docs/csharp/language-reference/operators/index.md:78]: I'm almost certain this is correct. I'll resolve this conversation.

**Review by BillWagner — APPROVED** [review]
Thanks again @pkulikov 

This is awesome. I'll :shipit: now.

**Pull request merged**